<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>Microarry pre-processing for Illumina BeadArray data</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: rgb(88, 72, 246)
   }

   pre .number {
     color: rgb(0, 0, 205);
   }

   pre .comment {
     color: rgb(76, 136, 107);
   }

   pre .keyword {
     color: rgb(0, 0, 255);
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: rgb(3, 106, 7);
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>




</head>

<body>
<h1>Microarry pre-processing for Illumina BeadArray data</h1>

<h2>Project Directory and Files</h2>

<pre><code class="r">
# Data Directory
project_dir=/media/D/expression/GAP_Expression

# Data Files
control_probe_profile_Final_Report.txt
Group_Probe_Profile_Final_Report.txt
probe_annotation_Final_Report.txt
Sample_and_Control_Probe_Profile_FinalReport.txt # input for lumiR
sample_table_Final_Report.txt
pheno_info.txt
batch_info.txt

</code></pre>

<h1>Begin R</h1>

<p>Start R and call the following</p>

<pre><code class="r">rm(list = ls())
options(stringsAsFactors = FALSE)
</code></pre>

<h2>setwd</h2>

<p>This is the full path to the directory where all the raw genomestudio output is stored</p>

<pre><code class="r">setwd(&quot;/media/D/sjnewhouse/GENE_EXPRESSION&quot;)
</code></pre>

<h2>load libs</h2>

<pre><code class="r"># Load libraries
library(lumi)
</code></pre>

<pre><code>## Loading required package: Biobase
## Loading required package: BiocGenerics
## Loading required package: parallel
## 
## Attaching package: &#39;BiocGenerics&#39;
## 
## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
## 
## The following object is masked from &#39;package:stats&#39;:
## 
##     xtabs
## 
## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, as.vector, cbind,
##     colnames, duplicated, eval, evalq, Filter, Find, get,
##     intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rep.int, rownames, sapply, setdiff, sort,
##     table, tapply, union, unique, unlist
## 
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.
</code></pre>

<pre><code>## Warning: replacing previous import by &#39;graphics::image&#39; when loading
## &#39;methylumi&#39;
</code></pre>

<pre><code>## KernSmooth 2.23 loaded
## Copyright M. P. Wand 1997-2009
</code></pre>

<pre><code>## Warning: replacing previous import by &#39;nleqslv::nleqslv&#39; when loading
## &#39;lumi&#39;
</code></pre>

<pre><code class="r">library(annotate)
</code></pre>

<pre><code>## Loading required package: AnnotationDbi
</code></pre>

<pre><code class="r">library(lumiHumanAll.db)
</code></pre>

<pre><code>## Loading required package: org.Hs.eg.db
## Loading required package: DBI
## 
## 
## lumiHumanAll.db is using or is likely to need access to special
##   nuID identifiers.  Users can learn about these identifiers from
##   vignette documentation provided with the lumi package.
</code></pre>

<pre><code class="r">library(affy)
</code></pre>

<pre><code>## 
## Attaching package: &#39;affy&#39;
## 
## The following objects are masked from &#39;package:lumi&#39;:
## 
##     MAplot, plotDensity
</code></pre>

<pre><code class="r">library(cluster)
library(impute)
library(WGCNA)
</code></pre>

<pre><code>## Loading required package: dynamicTreeCut
## Loading required package: flashClust
## 
## Attaching package: &#39;flashClust&#39;
## 
## The following object is masked from &#39;package:stats&#39;:
## 
##     hclust
## 
## Loading required package: Hmisc
## Loading required package: grid
## Loading required package: lattice
## Loading required package: survival
## Loading required package: splines
## Loading required package: Formula
## Hmisc library by Frank E Harrell Jr
## 
## Type library(help=&#39;Hmisc&#39;), ?Overview, or ?Hmisc.Overview&#39;)
## to see overall documentation.
## 
## 
## Attaching package: &#39;Hmisc&#39;
## 
## The following object is masked from &#39;package:survival&#39;:
## 
##     untangle.specials
## 
## The following object is masked from &#39;package:AnnotationDbi&#39;:
## 
##     contents
## 
## The following object is masked from &#39;package:lumi&#39;:
## 
##     combine
## 
## The following objects are masked from &#39;package:Biobase&#39;:
## 
##     combine, contents
## 
## The following object is masked from &#39;package:BiocGenerics&#39;:
## 
##     combine
## 
## The following objects are masked from &#39;package:base&#39;:
## 
##     format.pval, round.POSIXt, trunc.POSIXt, units
</code></pre>

<pre><code>## ==========================================================================
## *
## *  Package WGCNA 1.34 loaded.
## *
## *    Important note: It appears that your system supports multi-threading,
## *    but it is not enabled within WGCNA in R. 
## *    To allow multi-threading within WGCNA with all available cores, use 
## *
## *          allowWGCNAThreads()
## *
## *    within R. Use disableWGCNAThreads() to disable threading if necessary.
## *    Alternatively, set the following environment variable on your system:
## *
## *          ALLOW_WGCNA_THREADS=&lt;number_of_processors&gt;
## *
## *    for example 
## *
## *          ALLOW_WGCNA_THREADS=8
## *
## *    To set the environment variable in linux bash shell, type 
## *
## *           export ALLOW_WGCNA_THREADS=8
## *
## *     before running R. Other operating systems or shells will
## *     have a similar command to achieve the same aim.
## *
## ==========================================================================
</code></pre>

<pre><code>## 
## Attaching package: &#39;WGCNA&#39;
## 
## The following object is masked from &#39;package:stats&#39;:
## 
##     cor
</code></pre>

<pre><code class="r">library(gplots)
</code></pre>

<pre><code>## 
## Attaching package: &#39;gplots&#39;
## 
## The following object is masked from &#39;package:stats&#39;:
## 
##     lowess
</code></pre>

<pre><code class="r">library(limma)
</code></pre>

<pre><code>## 
## Attaching package: &#39;limma&#39;
## 
## The following object is masked from &#39;package:BiocGenerics&#39;:
## 
##     plotMA
</code></pre>

<pre><code class="r">library(vsn)
library(MBCB)
</code></pre>

<pre><code>## Loading required package: tcltk
## Loading required package: tcltk2
## 
## Attaching package: &#39;tcltk2&#39;
## 
## The following objects are masked from &#39;package:Hmisc&#39;:
## 
##     label, label&lt;-
</code></pre>

<pre><code class="r">library(lumiHumanIDMapping)
library(scatterplot3d)
library(relaimpo)
</code></pre>

<pre><code>## Loading required package: MASS
## 
## Attaching package: &#39;MASS&#39;
## 
## The following object is masked from &#39;package:AnnotationDbi&#39;:
## 
##     select
## 
## Loading required package: boot
## 
## Attaching package: &#39;boot&#39;
## 
## The following object is masked from &#39;package:survival&#39;:
## 
##     aml
## 
## The following object is masked from &#39;package:lattice&#39;:
## 
##     melanoma
## 
## Loading required package: survey
## 
## Attaching package: &#39;survey&#39;
## 
## The following object is masked from &#39;package:Hmisc&#39;:
## 
##     deff
## 
## The following object is masked from &#39;package:graphics&#39;:
## 
##     dotchart
## 
## Loading required package: mitools
## Loading required package: foreign
## This is the global version of package relaimpo.
## 
## If you are a non-US user, a version with the interesting additional metric pmvd is available
## 
## from Ulrike Groempings web site at prof.beuth-hochschule.de/groemping.
</code></pre>

<h2>source processing functions</h2>

<pre><code class="r"># path to gene expression processing scripts
path_to_scripts &lt;- &quot;/media/D/sjnewhouse/GENE_EXPRESSION&quot;
source(paste(path_to_scripts, &quot;/sjnewhouse_misc_R.R&quot;, sep = &quot;&quot;))
ls()
</code></pre>

<pre><code>##  [1] &quot;basic_sampleNetwork&quot;        &quot;basic_sampleNetworkIterate&quot;
##  [3] &quot;data_summary_plots&quot;         &quot;has_var_probe&quot;             
##  [5] &quot;has_var_probe2&quot;             &quot;max_probe&quot;                 
##  [7] &quot;mean_probe&quot;                 &quot;min_probe&quot;                 
##  [9] &quot;negBeadOutlierRepMean&quot;      &quot;path_to_scripts&quot;           
## [11] &quot;quantfun&quot;                   &quot;removeSamples_eset_lumi&quot;   
## [13] &quot;sd_probe&quot;                   &quot;shuffle_cols&quot;              
## [15] &quot;shuffle_rows&quot;               &quot;var_probe&quot;                 
## [17] &quot;write_expression_files&quot;     &quot;zero_var_probe&quot;
</code></pre>

<h2>set project settings and I/O</h2>

<p>User is asked to manually provide the options</p>

<pre><code class="r"># project directory
project_dir &lt;- &quot;/media/D/expression/GAP_Expression&quot;

# set working dir again
setwd(project_dir)

# project name
project_name &lt;- &quot;GAPtest01&quot;

# output directory for lumi process and plots
out_dir &lt;- paste(project_name, &quot;_lumi_processing&quot;, sep = &quot;&quot;)

# make project pre-processing directory
make_dir_command &lt;- paste(&quot; if [ ! -e ./&quot;, out_dir, &quot; ]; then mkdir ./&quot;, out_dir, 
    &quot;; fi&quot;, sep = &quot;&quot;)

system(make_dir_command)

# genomestudio reports
gs_report &lt;- &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/Sample_and_Control_Probe_Profile_FinalReport.txt&quot;
gs_probe &lt;- &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/Group_Probe_Profile_Final_Report.txt&quot;  # genomestudio report
gs_sample &lt;- &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/sample_table_Final_Report.txt&quot;  # genomestudio report
gs_control &lt;- &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/control_probe_profile_Final_Report.txt&quot;  # genomestudio report
anno_table &lt;- &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/probe_annotation_Final_Report.txt&quot;

# sample information FILE NAME must contain :
# Sample.ID,SEX,GROUPS,TISSUE,PHENOTYPE,Study_ID
pheno_file &lt;- &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/pheno_info.txt&quot;

# batch information
tech_pheno_file &lt;- &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/batch_info.txt&quot;

# detection call rate threshold
probe_det &lt;- 80
sample_det &lt;- 80

# flag for gender and sampleNetwork
sex_check &lt;- 1  # DO THIS!! I&#39;m not providing an option to skip this
iac_check &lt;- 1  # DO THIS!! I&#39;m not providing an option to skip this
iac_sd_thrs &lt;- 2  # 2 or 3

# Model based background correction method (MLE as default) All data should
# be background correceted. The recomended methods is MBCB (Model-based
# Background Correction for Beadarray) URL
# http://www.bioconductor.org/packages/release/bioc/html/MBCB.html
mbcb_method &lt;- &quot;MLE&quot;

# Transformation method
transform_method &lt;- &quot;log2&quot;  ## &#39;vst&#39; # log2, vst or both

# Normalisation method
norm_method &lt;- &quot;quantile&quot;  ## &#39;rsn&#39; # quantile, rsn, or both

# write settings to file
project_settings &lt;- data.frame(project_dir = project_dir, project_name = project_name, 
    out_dir = out_dir, gs_report = gs_report, gs_probe = gs_probe, gs_sample = gs_sample, 
    gs_control = gs_control, anno_table = anno_table, pheno_file = pheno_file, 
    tech_pheno_file = tech_pheno_file, probe_det = probe_det, sample_det = sample_det, 
    sex_check = sex_check, iac_check = iac_check, iac_sd_thrs = iac_sd_thrs, 
    mbcb_method = mbcb_method, transform_method = transform_method, norm_method = norm_method)

# write table to out_dir
write.table(project_settings, file = paste(out_dir, &quot;/&quot;, project_name, &quot;.project_settings&quot;, 
    sep = &quot;&quot;), row.names = FALSE, quote = FALSE, sep = &quot;\t&quot;)

# check settings
t(project_settings)
</code></pre>

<pre><code>##                  [,1]                                                                                                            
## project_dir      &quot;/media/D/expression/GAP_Expression&quot;                                                                            
## project_name     &quot;GAPtest01&quot;                                                                                                     
## out_dir          &quot;GAPtest01_lumi_processing&quot;                                                                                     
## gs_report        &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/Sample_and_Control_Probe_Profile_FinalReport.txt&quot;
## gs_probe         &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/Group_Probe_Profile_Final_Report.txt&quot;            
## gs_sample        &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/sample_table_Final_Report.txt&quot;                   
## gs_control       &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/control_probe_profile_Final_Report.txt&quot;          
## anno_table       &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/probe_annotation_Final_Report.txt&quot;               
## pheno_file       &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/pheno_info.txt&quot;                                  
## tech_pheno_file  &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/batch_info.txt&quot;                                  
## probe_det        &quot;80&quot;                                                                                                            
## sample_det       &quot;80&quot;                                                                                                            
## sex_check        &quot;1&quot;                                                                                                             
## iac_check        &quot;1&quot;                                                                                                             
## iac_sd_thrs      &quot;2&quot;                                                                                                             
## mbcb_method      &quot;MLE&quot;                                                                                                           
## transform_method &quot;log2&quot;                                                                                                          
## norm_method      &quot;quantile&quot;
</code></pre>

<h1>BEGIN PRE-PROCESSING</h1>

<h2>Raw Expression Set</h2>

<h2>1. read raw gene expression data</h2>

<pre><code class="r">
# raw input
gs_report
</code></pre>

<pre><code>## [1] &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/Sample_and_Control_Probe_Profile_FinalReport.txt&quot;
</code></pre>

<pre><code class="r">
# read raw gene expression data from genomestudio reports and create
# ExpressionSet

eset_raw &lt;- lumiR(paste(gs_report), lib.mapping = &quot;lumiHumanIDMapping&quot;, annotationColumn = c(&quot;PROBE_ID&quot;, 
    &quot;CHROMOSOME&quot;, &quot;SYMBOL&quot;, &quot;DEFINITION&quot;, &quot;ACCESSION&quot;, &quot;ENTREZ_GENE_ID&quot;, &quot;PROBE_TYPE&quot;, 
    &quot;PROBE_START&quot;, &quot;PROBE_SEQUENCE&quot;, &quot;PROBE_CHR_ORIENTATION&quot;, &quot;PROBE_COORDINATES&quot;, 
    &quot;CHROMOSOME&quot;, &quot;TRANSCRIPT&quot;, &quot;ILMN_GENE&quot;, &quot;REFSEQ_ID&quot;, &quot;UNIGENE_ID&quot;, &quot;SYMBOL&quot;, 
    &quot;PROTEIN_PRODUCT&quot;))
</code></pre>

<pre><code>## Inputting the data ...
## Perform Quality Control assessment of the LumiBatch object ...
## Directly converting probe sequence to nuIDs ...
</code></pre>

<pre><code class="r">
# check it
eset_raw
</code></pre>

<pre><code>## Summary of data information:
##   Data File Information:
##      GSGX Version    1.9.0
##      Report Date 29/10/2013 14:56:38
##      Project BRC_GAP_Expression_02
##      Group Set   BRC_GAP_Expression
##      Analysis    BRC_GAP_Expression_nonorm_nobkgd
##      Normalization   none
## 
## Major Operation History:
##             submitted            finished
## 1 2014-01-08 15:59:20 2014-01-08 16:05:06
## 2 2014-01-08 15:59:20 2014-01-08 16:05:06
##                                                                                                                                       command
## 1                    lumiR(&quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/Sample_and_Control_Probe_Profile_FinalReport.txt&quot;, 
## 2     annotationColumn = c(&quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/Sample_and_Control_Probe_Profile_FinalReport.txt&quot;, 
##   lumiVersion
## 1      2.14.1
## 2      2.14.1
## ...
##             submitted            finished
## 7 2014-01-08 16:05:06 2014-01-08 16:09:59
## 8 2014-01-08 16:09:59 2014-01-08 16:10:05
##                                                                       command
## 7        lumiQ(x.lumi = x.lumi, detectionTh = detectionTh, verbose = verbose)
## 8 addNuID2lumi(x.lumi = x.lumi, lib.mapping = lib.mapping, verbose = verbose)
##   lumiVersion
## 7      2.14.1
## 8      2.14.1
## 
## Object Information:
## LumiBatch (storageMode: lockedEnvironment)
## assayData: 47231 features, 618 samples 
##   element names: beadNum, detection, exprs, se.exprs 
## protocolData: none
## phenoData
##   sampleNames: 9020374058_A 9020374058_B ... 9249907052_L (618
##     total)
##   varLabels: sampleID
##   varMetadata: labelDescription
## featureData
##   featureNames: Ku8QhfS0n_hIOABXuE fqPEquJRRlSVSfL.8A ...
##     N8t5EuJCr0Tk9.zHno (47231 total)
##   fvarLabels: ProbeID TargetID ... DEFINITION (17 total)
##   fvarMetadata: labelDescription
## experimentData: use &#39;experimentData(object)&#39;
## Annotation: lumiHumanAll.db 
## Control Data: Available
## QC information: Please run summary(x, &#39;QC&#39;) for details!
</code></pre>

<h2>2. read in sample and batch information</h2>

<pre><code class="r">
# gs_sample
if (is.na(gs_sample)) stop(&quot; WARNING!: YOU HAVENT PROVIDED ANY SAMPLE INFORMATION!!!&quot;)
gs_sample
</code></pre>

<pre><code>## [1] &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/sample_table_Final_Report.txt&quot;
</code></pre>

<pre><code class="r">
raw_sample_data &lt;- read.table(paste(gs_sample), skip = 8, as.is = T, fill = T, 
    head = T, sep = &quot;\t&quot;)
rownames(raw_sample_data) &lt;- raw_sample_data$Sample.ID
raw_sample_data &lt;- raw_sample_data[, names(raw_sample_data) != &quot;X&quot;]  # added this as genomestudio likes to add mystery columns to the end of this report 
raw_n_samples &lt;- dim(raw_sample_data)[1]  # number of rows ie samples

# pheno_file
if (is.na(pheno_file)) stop(&quot; WARNING!: YOU HAVENT PROVIDED ANY PHENOTYPE INFORMATION!!!&quot;)
pheno_file
</code></pre>

<pre><code>## [1] &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/pheno_info.txt&quot;
</code></pre>

<pre><code class="r">
pheno_dat &lt;- read.table(paste(pheno_file), as.is = T, fill = T, head = T, sep = &quot;\t&quot;)
</code></pre>

<pre><code>## Warning: cannot open file
## &#39;/media/D/expression/GAP_Expression/final_reports_genomestudio/pheno_info.txt&#39;:
## No such file or directory
</code></pre>

<pre><code>## Error: cannot open the connection
</code></pre>

<pre><code class="r">has_pheno_cols &lt;- c(&quot;Sample.ID&quot;, &quot;SEX&quot;, &quot;GROUPS&quot;, &quot;TISSUE&quot;, &quot;PHENOTYPE&quot;, &quot;Study_ID&quot;) %in% 
    names(pheno_dat)
</code></pre>

<pre><code>## Error: object &#39;pheno_dat&#39; not found
</code></pre>

<pre><code class="r">missing_pheno_cols &lt;- &quot;FALSE&quot; %in% has_pheno_cols
</code></pre>

<pre><code>## Error: object &#39;has_pheno_cols&#39; not found
</code></pre>

<pre><code class="r">if (missing_pheno_cols == &quot;TRUE&quot;) stop(&quot; WARNING!: YOU ARE MISSING ESSENTIAL SAMPLE INFORMATION! MAKE SURE YOUR PHENO_FILE HAS:- Sample.ID,SEX,GROUPS,TISSUE,PHENOTYPE,Study_ID !!!&quot;)
</code></pre>

<pre><code>## Error: object &#39;missing_pheno_cols&#39; not found
</code></pre>

<pre><code class="r">raw_n_pheno_dat &lt;- dim(pheno_dat)[1]  # number of rows ie samples
</code></pre>

<pre><code>## Error: object &#39;pheno_dat&#39; not found
</code></pre>

<pre><code class="r">
# tech_pheno_file
if (is.na(tech_pheno_file)) stop(&quot; WARNING!: YOU HAVENT PROVIDED ANY BATCH INFORMATION!!!&quot;)
tech_pheno_file
</code></pre>

<pre><code>## [1] &quot;/media/D/expression/GAP_Expression/final_reports_genomestudio/batch_info.txt&quot;
</code></pre>

<pre><code class="r">
tech_pheno &lt;- read.table(paste(tech_pheno_file), head = T, sep = &quot;\t&quot;)
tech_pheno$Sentrix.Barcode &lt;- as.character(tech_pheno$Sentrix.Barcode)
colnames(tech_pheno) &lt;- paste(&quot;tech.&quot;, names(tech_pheno), sep = &quot;&quot;)

# pData()
eset_samples &lt;- pData(eset_raw)


# quick check these should all have the same number of rows or samples!
dim(eset_samples)
</code></pre>

<pre><code>## [1] 618   1
</code></pre>

<pre><code class="r">dim(gs_sample)
</code></pre>

<pre><code>## NULL
</code></pre>

<pre><code class="r">dim(pheno_dat)
</code></pre>

<pre><code>## Error: object &#39;pheno_dat&#39; not found
</code></pre>

<pre><code class="r">dim(tech_pheno)
</code></pre>

<pre><code>## [1] 641  17
</code></pre>

<pre><code class="r">
# col names
names(eset_samples)
</code></pre>

<pre><code>## [1] &quot;sampleID&quot;
</code></pre>

<pre><code class="r">names(gs_sample)
</code></pre>

<pre><code>## NULL
</code></pre>

<pre><code class="r">names(pheno_dat)
</code></pre>

<pre><code>## Error: object &#39;pheno_dat&#39; not found
</code></pre>

<pre><code class="r">names(tech_pheno)
</code></pre>

<pre><code>##  [1] &quot;tech.Sample.ID&quot;                      
##  [2] &quot;tech.Sentrix.Barcode&quot;                
##  [3] &quot;tech.SampleSection&quot;                  
##  [4] &quot;tech.Batch&quot;                          
##  [5] &quot;tech.Date_out&quot;                       
##  [6] &quot;tech.Date_extraction&quot;                
##  [7] &quot;tech.person&quot;                         
##  [8] &quot;tech.Conc_Nanodrop&quot;                  
##  [9] &quot;tech.Date_Dilutionand_Amplification&quot; 
## [10] &quot;tech.Date_cRNApurification&quot;          
## [11] &quot;tech.Date_Quantitation_by_RiboGreen&quot; 
## [12] &quot;tech.Eluted_Total_labelled_cRNA&quot;     
## [13] &quot;tech.labelled_cRNA_Yield&quot;            
## [14] &quot;tech.concentration_of_labelled_cRNA&quot; 
## [15] &quot;tech.Date_labelled_cRNA&quot;             
## [16] &quot;tech.Date_Hybridization_for_15_hours&quot;
## [17] &quot;tech.Date_Washing_and_scanning&quot;
</code></pre>

<h2>3. check pData(), sample &amp; batch info Sample.ID&#39;s match in names and numbers</h2>

<pre><code class="r"># check they all match
raw_n_samples == raw_n_pheno_dat
</code></pre>

<pre><code>## Error: object &#39;raw_n_pheno_dat&#39; not found
</code></pre>

<h2>4. Update pData() slot and subset raw ExpressionSet to matched/complete Sample.IDs</h2>

<h2>5. Save raw ExpressionSet eset_raw</h2>

<h2>6. Write data files to out_dir for eset_raw</h2>

<h2>7. Basic QC and plots on eset_raw</h2>

<p>##</p>

<h2>MBCB (Model-based Background Correction for Beadarray)</h2>

</body>

</html>

