Microarry pre-processing for Illumina BeadArray data
=====================================================

## Project Directory and Files
```{r Files, eval=FALSE}

# Data Directory
project_dir=/media/D/expression/GAP_Expression

# Data Files
control_probe_profile_Final_Report.txt
Group_Probe_Profile_Final_Report.txt
probe_annotation_Final_Report.txt
Sample_and_Control_Probe_Profile_FinalReport.txt # input for lumiR
sample_table_Final_Report.txt
pheno_info.txt
batch_info.txt

```

# Begin R
Start R and call the following

```{r begin, eval=TRUE}
rm(list=ls())
options(stringsAsFactors = FALSE)
```

## setwd
This is the full path to the directory where all the raw genomestudio output is stored

```{r, setwd, eval=TRUE}
setwd("/media/D/sjnewhouse/GENE_EXPRESSION")
```

## load libs
```{r loadLibs, eval=TRUE}
# Load libraries 
library(lumi)
library(annotate)
library(lumiHumanAll.db)
library(affy)
library(cluster)
library(impute)
library(WGCNA)
library(gplots)
library(limma)
library(vsn)
library(MBCB)
library(lumiHumanIDMapping)
library(scatterplot3d)
library(relaimpo)
```

## source processing functions
```{r loadSource, eval=TRUE}
# path to gene expression processing scripts
path_to_scripts <- "/media/D/sjnewhouse/GENE_EXPRESSION" 
source(paste(path_to_scripts,"/sjnewhouse_misc_R.R",sep=""))
ls()
```

## set project settings and I/O
User is asked to manually provide the options

```{r setProjectOptions, eval=TRUE}
# project directory
project_dir <- "/media/D/expression/GAP_Expression"

# set working dir again
setwd(project_dir)

# project name
project_name <- "GAPtest01"

# output directory for lumi process and plots
out_dir <- paste(project_name,"_lumi_processing" ,sep="")

# make project pre-processing directory
make_dir_command <- paste(" if [ ! -e ./",out_dir," ]; then mkdir ./",out_dir,"; fi",sep="")

system( make_dir_command )

# genomestudio reports
gs_report <- "/media/D/expression/GAP_Expression/final_reports_genomestudio/Sample_and_Control_Probe_Profile_FinalReport.txt"
gs_probe <- "/media/D/expression/GAP_Expression/final_reports_genomestudio/Group_Probe_Profile_Final_Report.txt" # genomestudio report
gs_sample <- "/media/D/expression/GAP_Expression/final_reports_genomestudio/sample_table_Final_Report.txt" # genomestudio report
gs_control <- "/media/D/expression/GAP_Expression/final_reports_genomestudio/control_probe_profile_Final_Report.txt" # genomestudio report
anno_table <- "/media/D/expression/GAP_Expression/final_reports_genomestudio/probe_annotation_Final_Report.txt"

# sample information
# FILE NAME must contain : Sample.ID,SEX,GROUPS,TISSUE,PHENOTYPE,Study_ID
pheno_file <- "/media/D/expression/GAP_Expression/final_reports_genomestudio/pheno_info.txt" 

# batch information
tech_pheno_file <-"/media/D/expression/GAP_Expression/final_reports_genomestudio/batch_info.txt"

# detection call rate threshold
probe_det <- 80
sample_det <- 80

# flag for gender and sampleNetwork 
sex_check <- 1   # DO THIS!! I'm not providing an option to skip this
iac_check <- 1   # DO THIS!! I'm not providing an option to skip this
iac_sd_thrs <- 2 # 2 or 3

# Model based background correction method (MLE as default)
# All data should be background correceted. The recomended methods is MBCB (Model-based Background Correction for Beadarray)
# URL http://www.bioconductor.org/packages/release/bioc/html/MBCB.html
mbcb_method <- "MLE"

# Transformation method
transform_method <- "log2" ## "vst" # log2, vst or both

# Normalisation method
norm_method <- "quantile" ## "rsn" # quantile, rsn, or both

# write settings to file
project_settings <- data.frame(
  project_dir=project_dir,
  project_name=project_name,
  out_dir=out_dir,
  gs_report=gs_report,
  gs_probe=gs_probe,
  gs_sample=gs_sample,
  gs_control=gs_control,
  anno_table=anno_table,
  pheno_file=pheno_file,
  tech_pheno_file=tech_pheno_file,
  probe_det=probe_det,
  sample_det=sample_det,
  sex_check=sex_check,
  iac_check=iac_check,
  iac_sd_thrs=iac_sd_thrs,
  mbcb_method=mbcb_method,
  transform_method=transform_method,
  norm_method=norm_method  
  )

# write table to out_dir
write.table(project_settings, file=paste(out_dir,"/",project_name,".project_settings",sep=""),row.names=FALSE,quote=FALSE,sep="\t")

# check settings
t(project_settings)

```

BEGIN PRE-PROCESSING
=====================

Raw Expression Set
-------------------

## 1. read raw gene expression data 
```{r lumiR, eval=TRUE}

# raw input 
gs_report 

# read raw gene expression data from genomestudio reports and create ExpressionSet

eset_raw <- lumiR(paste(gs_report), lib.mapping="lumiHumanIDMapping",annotationColumn = c('PROBE_ID','CHROMOSOME','SYMBOL','DEFINITION','ACCESSION','ENTREZ_GENE_ID','PROBE_TYPE','PROBE_START','PROBE_SEQUENCE','PROBE_CHR_ORIENTATION','PROBE_COORDINATES','CHROMOSOME','TRANSCRIPT','ILMN_GENE','REFSEQ_ID','UNIGENE_ID','SYMBOL','PROTEIN_PRODUCT'))

# check it
eset_raw

```

## 2. read in sample and batch information
```{r readSamplePhenoAndBatchInfo, eval=TRUE}

# gs_sample
if(is.na(gs_sample))  stop(" WARNING!: YOU HAVENT PROVIDED ANY SAMPLE INFORMATION!!!" )
gs_sample

raw_sample_data <- read.table(paste(gs_sample) ,skip=8,as.is=T,fill=T,head=T,sep="\t")
rownames(raw_sample_data) <- raw_sample_data$Sample.ID
raw_sample_data <- raw_sample_data[,names(raw_sample_data)!="X"] # added this as genomestudio likes to add mystery columns to the end of this report 
raw_n_samples <- dim(raw_sample_data)[1]  # number of rows ie samples
save(raw_sample_data,file=paste(out_dir,"/",project_name,".raw_sample_data.RData",sep="") )

#  pheno_file
if(is.na(pheno_file))  stop(" WARNING!: YOU HAVENT PROVIDED ANY PHENOTYPE INFORMATION!!!" )
pheno_file

pheno_dat <- read.table(paste(pheno_file), as.is=T,fill=T,head=T,sep="\t")
save(pheno_dat,file=paste(out_dir,"/",project_name,".pheno_dat.RData",sep="") )
has_pheno_cols <- c("Sample.ID","SEX","GROUPS","TISSUE","PHENOTYPE","Study_ID") %in% names(pheno_dat);
missing_pheno_cols <- "FALSE" %in% has_pheno_cols
if(missing_pheno_cols == "TRUE") stop(" WARNING!: YOU ARE MISSING ESSENTIAL SAMPLE INFORMATION! MAKE SURE YOUR PHENO_FILE HAS:- Sample.ID,SEX,GROUPS,TISSUE,PHENOTYPE,Study_ID !!!")
raw_n_pheno_dat <- dim(pheno_dat)[1] # number of rows ie samples

# tech_pheno_file
if(is.na(tech_pheno_file))  stop(" WARNING!: YOU HAVENT PROVIDED ANY BATCH INFORMATION!!!" )
tech_pheno_file

tech_pheno <- read.table(paste(tech_pheno_file),head=T,sep="\t")
tech_pheno$Sentrix.Barcode <- as.character(tech_pheno$Sentrix.Barcode)
colnames(tech_pheno) <- paste("tech.",names(tech_pheno),sep="")
save(tech_pheno,file=paste(out_dir,"/",project_name,".tech_pheno.RData",sep="") )

# get pData()
eset_samples <- pData(eset_raw)
# add chip order and flad for "has expression data" to eset_samples (pData)
eset_samples$has_expression <- 1
eset_samples$chip_order <- 1:dim(eset_samples)[1]
save(eset_samples,file=paste(out_dir,"/",project_name,".eset_samples.RData",sep="") )

# quick check
# these should all have the same number of rows or samples!
dim(eset_samples)
dim(raw_sample_data) 
dim(pheno_dat)
dim(tech_pheno)

# col names
names(eset_samples)
names(raw_sample_data)
names(pheno_dat)
names(tech_pheno)

# Venn of Sample.ID
ex <- eset_samples$sampleID
pp <- pheno_dat$Sample.ID
tt <- tech_pheno$tech.Sample.ID
gs <- raw_sample_data$Sample.ID
venninput  <-list(ArrayExpression=ex,Batch_Info=tt,Pheno_Info=pp,GenomeStudio=gs)
venn(venninput)

```

## 3. check eset_samples, sample & batch info Sample.ID's match in names and numbers & merge all
```{r MatchSamplePhenoAndBatchInfo, eval=TRUE}

# 1. merge eset_samples with pheno_dat. Keep ALL overlaps only
eset_pheno_merge <- merge(eset_samples,pheno_dat,by.x.="sampleID",by.y="Sample.ID" )
eset_pheno_merge <- eset_pheno_merge[order(eset_pheno_merge$chip_order),]
dim(eset_samples);eset_pheno_merge # check size

# 2. merge eset_pheno_merge with tech_pheno
eset_pheno_batch_merge <- merge(eset_pheno_merge,tech_pheno,by.x.="Sample.ID",by.y="tech.Sample.ID" )
eset_pheno_batch_merge <- eset_pheno_batch_merge[order(eset_pheno_batch_merge$chip_order),]
dim(eset_samples);dim(eset_pheno_merge);dim(eset_pheno_batch_merge) # check size

# 3. merge all with genomestudio final report
eset_pheno_batch_gs_merge <- merge(eset_pheno_batch_merge,gs_sample,by.x.="Sample.ID",by.y="Sample.ID" )
eset_pheno_batch_gs_merge <- eset_pheno_batch_gs_merge[order(eset_pheno_batch_gs_merge$chip_order),]

dim(eset_samples);dim(eset_pheno_merge);dim(eset_pheno_batch_merge);dim(eset_pheno_batch_gs_merge) # check size





```

## 4. Update pData() slot and subset raw ExpressionSet to matched/complete Sample.IDs
```{r subsetEsetRaw, eval=TRUE}




```

## 5. Save raw ExpressionSet eset_raw
```{r saveEsetRaw, eval=TRUE}

```

## 6. Write data files to out_dir for eset_raw
```{r writeEsetRawData, eval=TRUE}

```

## 7. Basic QC and plots on eset_raw
```{r basicQC_eset_raw, eval=TRUE}

```



##
```{r}

```

MBCB (Model-based Background Correction for Beadarray)
-------------------------------------------------------




