#!/share/bin/Rscript
rm(list=ls())

#############################################################################
## Authors : Stephen Newhouse, Amos Forlarin
## Email : stephen..j.newhouse@gmail.comp
## Date : 15 Nov 2013
##############################################################################
##
## GX ILLUMINA PIPELINE : Intensity QC
##
## Verison: 1.00
##
## This ids poor samples based on intensity/technical data generated by 
## Genomestudio. Its is expected that you run this 1st as part of the 
## NIHR BRC-MH Illumina Gene Expression Pipeline,
## prior to imputing missing probes and generating the *Final_Reports*
## The samples id'd as part of this qc, should be removed from your project
## and a new project created using the "good" samples.
##
## see SOP [ADD URL]
##
## USAGE:
##
## Rscript gx_illumina_pipeline_intensity_qc.1.0.R <CONFIG_FILE>
##
##############################################################################

args <- commandArgs(trailingOnly=TRUE);

data_dir <- getwd()

cat(" Data Dir ",data_dir,"\r","\n")

cat(" Set Working Dir to ", data_dir,"\r","\n")

setwd(data_dir)

###########################################################################

cat(" Getting Project settings and file names","\r","\n")

config_file <- args[1];

cat(" Reading Config file ","\r","\n")

projectConf <- read.table(config_file,as.is=T,fill=T,head=T,sep="\t")

gs_sample <- projectConf$gs_sample ## "sample_table_Final_Report.txt"

cat(" Sample Table Final Report:- ", gs_sample ,"\r","\n")

project_name <- projectConf$project_name ## "GAP" 

cat(" Project Name:- ", project_name ,"\r","\n")

out_dir <- paste(project_name,"_lumi_processing" ,sep="") # output dir for lumi process and plots

z_threshold <- projectConf$z_threshold ## 2

cat(" Z SD Threshold:- ",z_threshold,"\n","\r")


cat(" Making processing directory. mkdir ./",out_dir,"\r","\n")

system( paste(" mkdir ./",out_dir,sep="") )

############################################################
## 1. Sample qc based on technical data from Genomestudio ##
############################################################

cat(" Reading Sample Table Final Report generated in Genomestudio ", gs_sample ,"\r","\n")

raw_sample_data <- read.table(gs_sample ,skip=8,as.is=T,fill=T,head=T,sep="\t")

raw_n_samples <- dim(raw_sample_data)[1]

cat(" Number of samples = ",raw_n_samples,"\r","\n")
cat(" Starting outlier samples based on intensity data for Deteced Genes (0.05), Signal Average, Signal P05 and Signal P95","\n","\r")
cat(" Scale Intensity data: as.numeric(scale(INTENSITY)) for Deteced Genes (0.05), Signal Average, Signal P05 and Signal P95 ", "\r","\n")

raw_sample_data$z_Detected.Genes.05 <- as.numeric(scale(raw_sample_data$Detected.Genes..0.05.))
raw_sample_data$z_Signal.Average <- as.numeric(scale(raw_sample_data$Signal.Average))
raw_sample_data$z_Signal.P05 <- as.numeric(scale(raw_sample_data$Signal.P05))
raw_sample_data$z_Signal.P95 <- as.numeric(scale(raw_sample_data$Signal.P95))

cat(" Find outlier samples based on scaled intensity data for Deteced Genes (0.05), Signal Average, Signal P05 and Signal P95 ", "\r","\n")
cat(" SD Threshold is ",z_threshold ,"\r","\n")

samp_out_1 <- raw_sample_data[abs(raw_sample_data$z_Detected.Genes.05)>"-2",]$Sample.ID;
samp_out_2 <- raw_sample_data[abs(raw_sample_data$z_Signal.Average)>z_threshold,]$Sample.ID;
samp_out_3 <- raw_sample_data[abs(raw_sample_data$z_Signal.P05)>z_threshold,]$Sample.ID;
samp_out_4 <- raw_sample_data[abs(raw_sample_data$z_Signal.P95)>z_threshold,]$Sample.ID;

intensity_outliers <- unique(c(samp_out_1,samp_out_2,samp_out_3,samp_out_4))

cat(" intensity_outliers sorted as object [intensity_outliers]","\r","\n")

n_intensity_outliers <- length(intensity_outliers)

p_intensity_outliers <- round(n_intensity_outliers/raw_n_samples,3)

iout <- data.frame(Sample.ID=intensity_outliers,QC=rep("intensity_outliers",n_intensity_outliers))

int_qc_samples <- paste(out_dir,"/",project_name,".intensity_outliers.txt",sep="")

cat(" Writing intensity outlier samples for Deteced Genes (0.05), Signal Average, Signal P05 and Signal P95 to ",int_qc_samples,"\r","\n")

write.table(iout,file=int_qc_samples,quote=F,row.names=F,sep="\t")

cat(" Number of outlier samples based on scaled intensity data for Deteced Genes (0.05), Signal Average, Signal P05 and Signal P95 : ",n_intensity_outliers,"/",raw_n_samples,":",p_intensity_outliers,"%" ,"\r","\n")

cat(" Making Plots of intensity data for Deteced Genes (0.05), Signal Average, Signal P05 and Signal P95","\r","\n")

z_cols <- grep("z_",names(raw_sample_data))
z_min <- min(raw_sample_data[,z_cols])
z_max <- max(raw_sample_data[,z_cols])

int_qc_plots <- paste(out_dir,"/",project_name,".intensity_outliers.pdf",sep="")

cat(" Writing plots to ",int_qc_plots,"\r","\n")

pdf(int_qc_plots, width=8, height=8)

plot( raw_sample_data$Index, raw_sample_data$z_Detected.Genes.05, main=paste(project_name," : Raw Expression :- Z Detected Genes (p=0.05)",sep="") , xlab="Sampel Index",ylab="Z Detected Genes (p=0.05)",ylim=c(floor(z_min),ceiling(z_max)))
points( raw_sample_data$Index[raw_sample_data$Sample.ID%in%samp_out_1], raw_sample_data$z_Detected.Genes.05[raw_sample_data$Sample.ID%in%samp_out_1 ],col="red")
abline(h=-2,lty=2)

plot( raw_sample_data$Index, raw_sample_data$z_Signal.Average, main=paste(project_name," : Raw Expression :- Z Signal Average ",sep="") , xlab="Sampel Index",ylab="Z Signal Average",ylim=c(floor(z_min),ceiling(z_max)))
points( raw_sample_data$Index[raw_sample_data$Sample.ID%in%samp_out_2], raw_sample_data$z_Signal.Average[raw_sample_data$Sample.ID%in%samp_out_2 ],col="red")
abline(h=-2,lty=2);abline(h=2,lty=2);

plot( raw_sample_data$Index, raw_sample_data$z_Signal.P05, main=paste(project_name," : Raw Expression :- Z Signal.P05",sep="") , xlab="Sampel Index",ylab="Z Signal.P05",ylim=c(floor(z_min),ceiling(z_max)))
points( raw_sample_data$Index[raw_sample_data$Sample.ID%in%samp_out_3], raw_sample_data$z_Signal.P05[raw_sample_data$Sample.ID%in%samp_out_3 ],col="red")
abline(h=-2,lty=2);abline(h=2,lty=2);

plot( raw_sample_data$Index, raw_sample_data$z_Signal.P95, main=paste(project_name," : Raw Expression :- Z Signal.P95",sep="") , xlab="Sampel Index",ylab="Z Signal.P95",ylim=c(floor(z_min),ceiling(z_max)))
points( raw_sample_data$Index[raw_sample_data$Sample.ID%in%samp_out_4], raw_sample_data$z_Signal.P95[raw_sample_data$Sample.ID%in%samp_out_4 ],col="red")
abline(h=-2,lty=2);abline(h=2,lty=2);

dev.off()

sampleF <- paste(out_dir,"/",project_name,".raw_sample_data.txt",sep="")


cat(" Writing new samples Data ",sampleF,"\r","\n")

write.table(raw_sample_data,file=sampleF,quote=F,row.names=F,sep="\t")

cat(" Done outlier samples based on intensity data for Deteced Genes (0.05), Signal Average, Signal P05 and Signal P95","\n","\r")

rm(list=ls())

gc()

q()

n



